chains/evm/contracts/test/onRamp/OnRamp/OnRamp.getReceipts.t.sol

```diff
@@ -0,0 +1,374 @@
+    );
+  }
+
+  function _createMessage(
+    uint256 tokenAmount
+  ) internal returns (Client.EVM2AnyMessage memory) {
+    Client.EVMTokenAmount[] memory tokenAmounts = new Client.EVMTokenAmount[](tokenAmount > 0 ? 1 : 0);
+    if (tokenAmount > 0) {
+      tokenAmounts[0] = Client.EVMTokenAmount({token: s_sourceToken, amount: tokenAmount});
+    }
+
+    return Client.EVM2AnyMessage({
+      receiver: abi.encodePacked(makeAddr("receiver")),
+      data: "test data",
+      tokenAmounts: tokenAmounts,
+      feeToken: s_sourceFeeToken,
+      extraArgs: ""
+    });
+  }
+
+  function _createExtraArgs(
+    address[] memory ccvAddresses
+  ) internal view returns (ExtraArgsCodec.GenericExtraArgsV3 memory) {
+    address[] memory ccvs = new address[](ccvAddresses.length);
+    bytes[] memory execArgs = new bytes[](ccvAddresses.length);
+    for (uint256 i = 0; i < ccvAddresses.length; i++) {
+      ccvs[i] = ccvAddresses[i];
+    }
+
+    return ExtraArgsCodec.GenericExtraArgsV3({
+      ccvs: ccvs,
+      ccvArgs: execArgs,
+      blockConfirmations: 0,
+      gasLimit: GAS_LIMIT,
+      executor: s_defaultExecutor,
+      executorArgs: "",
+      tokenReceiver: "",
+      tokenArgs: ""
+    });
+  }
+
+  function _mockVerifier(address verifierAddress, address implAddress) internal {
+    vm.mockCall(
+      verifierAddress,
+      abi.encodeCall(ICrossChainVerifierResolver.getOutboundImplementation, (DEST_CHAIN_SELECTOR, "")),
+      abi.encode(implAddress)
+    );
+  }
+
+  function _mockVerifierFee(
+    address implAddress
+  ) internal {
+    vm.mockCall(
+      implAddress,
+      abi.encodeWithSelector(ICrossChainVerifierV1.getFee.selector),
+      abi.encode(VERIFIER_FEE_USD_CENTS, VERIFIER_GAS, VERIFIER_BYTES)
+    );
+  }
+
+  function test_getReceipts_WithTokens_PoolV2Fee_Success() public {
+    // Mock pool to support IPoolV2.
+    vm.mockCall(s_pool, abi.encodeCall(IERC165.supportsInterface, (type(IPoolV2).interfaceId)), abi.encode(true));
+
+    // Mock pool's getFee to return custom fees.
+    vm.mockCall(
+      s_pool,
+      abi.encodeWithSelector(IPoolV2.getFee.selector),
+      abi.encode(POOL_FEE_USD_CENTS, POOL_GAS_OVERHEAD, POOL_BYTES_OVERHEAD, uint16(0))
+    );
+
+    // Mock verifiers.
+    address verifier1Impl = makeAddr("verifier1Impl");
+    address verifier2Impl = makeAddr("verifier2Impl");
+    _mockVerifier(s_verifier1, verifier1Impl);
+    _mockVerifier(s_verifier2, verifier2Impl);
+    _mockVerifierFee(verifier1Impl);
+    _mockVerifierFee(verifier2Impl);
+
+    Client.EVM2AnyMessage memory message = _createMessage(100 ether);
+    address[] memory ccvs = new address[](2);
+    ccvs[0] = s_verifier1;
+    ccvs[1] = s_verifier2;
+    ExtraArgsCodec.GenericExtraArgsV3 memory extraArgs = _createExtraArgs(ccvs);
+
+    OnRamp.Receipt[] memory receipts = s_onRampHelper.getReceipts(DEST_CHAIN_SELECTOR, message, extraArgs);
+
+    // Should have: 2 verifiers + 1 token + 1 executor = 4 receipts.
```

Comment:
Wondering if s_onRampHelper should expose helper functions for getVerifierReceipts / getExecutorReceipt / getTokenPoolReceipts. We should try to minimize impact to our tests in the case that ordering ever changes.